src/jsdoc_transformer.ts

```diff
@@ -0,0 +1,937 @@
+          // aware of the actual type hierarchy of the class.
+          continue;
+        }
+        tagName = 'extends';
+      } else if (alias.flags & ts.SymbolFlags.Value) {
+        // If the symbol was already in the value namespace, then it will
+        // not be a type in the Closure output (because Closure collapses
+        // the type and value namespaces).  Just ignore the implements.
+        continue;
+      }
+      // typeToClosure includes nullability modifiers, so call symbolToString directly here.
+      docTags.push({tagName, type: typeTranslator.symbolToString(alias, true)});
+    }
+  }
+}
+
+/** Flags that declare a field of the same name if set on a ctor parameter. */
+const FIELD_DECLARATION_MODIFIERS: ts.ModifierFlags = ts.ModifierFlags.Private |
+    ts.ModifierFlags.Protected | ts.ModifierFlags.Public | ts.ModifierFlags.Readonly;
+
+/**
+ * emitMemberTypes emits the type annotations for members of a class. It's necessary in the case
+ * where TypeScript syntax specifies there are additional properties on the class, because to
+ * declare these in Closure you must declare these separately from the class.
+ */
+function createMemberTypeDeclaration(
+    mtt: ModuleTypeTranslator,
+    typeDecl: ts.ClassDeclaration|ts.InterfaceDeclaration): ts.IfStatement|null {
+  // Gather parameter properties from the constructor, if it exists.
+  const ctors: ts.ConstructorDeclaration[] = [];
+  let paramProps: ts.ParameterDeclaration[] = [];
+  const nonStaticProps: Array<ts.PropertyDeclaration|ts.PropertySignature> = [];
+  const staticProps: Array<ts.PropertyDeclaration|ts.PropertySignature> = [];
+  const unhandled: ts.NamedDeclaration[] = [];
+  const abstractMethods: ts.FunctionLikeDeclaration[] = [];
+  for (const member of typeDecl.members) {
+    if (member.kind === ts.SyntaxKind.Constructor) {
+      ctors.push(member as ts.ConstructorDeclaration);
+    } else if (ts.isPropertyDeclaration(member) || ts.isPropertySignature(member)) {
+      const isStatic = transformerUtil.hasModifierFlag(member, ts.ModifierFlags.Static);
+      if (isStatic) {
+        staticProps.push(member);
+      } else {
+        nonStaticProps.push(member);
+      }
+    } else if (
+        member.kind === ts.SyntaxKind.MethodDeclaration ||
+        member.kind === ts.SyntaxKind.MethodSignature ||
+        member.kind === ts.SyntaxKind.GetAccessor || member.kind === ts.SyntaxKind.SetAccessor) {
+      if (transformerUtil.hasModifierFlag(member, ts.ModifierFlags.Abstract) ||
+          ts.isInterfaceDeclaration(typeDecl)) {
+        abstractMethods.push(
+            member as ts.MethodDeclaration | ts.GetAccessorDeclaration | ts.SetAccessorDeclaration);
+      }
+      // Non-abstract methods only exist on classes, and are handled in regular emit.
+    } else {
+      unhandled.push(member);
+    }
+  }
+
+  if (ctors.length > 0) {
+    const ctor = ctors[0];
```

Comment:
Is this choice arbitrary?

Reply:
You tell me, I think you wrote this code :-)

But actually, you can only have param props on the last ctor. Fixed the code and added a test.